display_list/dl_color.h

```diff
@@ -9,10 +9,32 @@
 
 namespace flutter {
 
+enum class DlColorSpace { kSRGB = 0, kExtendedSRGB = 1, kDisplayP3 = 2 };
+
 struct DlColor {
  public:
-  constexpr DlColor() : argb_(0xFF000000) {}
-  constexpr explicit DlColor(uint32_t argb) : argb_(argb) {}
+  constexpr DlColor()
+      : alpha_(0.f),
+        red_(0.f),
+        green_(0.f),
+        blue_(0.f),
+        color_space_(DlColorSpace::kSRGB) {}
+  constexpr explicit DlColor(uint32_t argb)
+      : alpha_(toF((argb >> 24) & 0xff)),
+        red_(toF((argb >> 16) & 0xff)),
+        green_(toF((argb >> 8) & 0xff)),
+        blue_(toF((argb >> 0) & 0xff)),
+        color_space_(DlColorSpace::kSRGB) {}
+  constexpr DlColor(DlScalar alpha,
```

Comment:
Now I remember. I didn't want to have a constructor that had an implicit ordering of the components so I created the ARGB and RGBA factories. If I'd my druthers I would also question the existence of the uint32_t constructor with its implicit byte order. How often was this constructor used and could we make it private and only used from the factories and the relatively small number of existing legacy constructors?

Reply:
the uint32_t constructor has been around since the class was created.

The RGBA and ARGB constructors were only created last month for this reason.

Reply:
We should probably also make an `ARGB(uint32_t)` single argument factory and start replacing the raw constructor everywhere.

Reply:
Removing that constructor sounds good to me.  I think that would be better suited for a different PR though since it wasn't introduced here.